src/devices/bus/isa/mda.cpp

```diff
@@ -965,3 +975,514 @@ MC6845_UPDATE_ROW( isa8_ec1840_0002_device::crtc_update_row )
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void isa8_epc_mda_device::device_start()
+{
+	if (m_palette != nullptr && !m_palette->started())
+		throw device_missing_dependencies();
+
+	/* Palette for use with the Ericsson Amber Monochrome HR CRT monitor 3111, P3 phospor 602nm 255,183, 0 */
+	m_3111_pal[0] = rgb_t(  0,   0,     0); // black
+	m_3111_pal[1] = rgb_t(  143, 103,   0); // dim
+	m_3111_pal[2] = rgb_t(  191, 137,   0); // normal
+	m_3111_pal[3] = rgb_t(  255, 183,   0); // bright
+
+	/* Palette for use with the Ericsson B&W Monochrome HR CRT monitor 3712/3715 */
+	m_371x_pal[0] = rgb_t(    0,   0,   0); // black
+	m_371x_pal[1] = rgb_t(  143, 143, 143); // dim
+	m_371x_pal[2] = rgb_t(  191, 191, 191); // normal
+	m_371x_pal[3] = rgb_t(  255, 255, 255); // bright
+
+	/* Init a default palette */
+	m_pal = &m_3111_pal; // In case screen starts rendering before device_reset where we read the settings
+
+	m_videoram.resize(0x8000);
+	set_isa_device();
+	m_installed = false;
+}
+
+void isa8_epc_mda_device::device_reset()
+{
+	m_framecnt = 0;
+	m_mode_control = 0;
+	m_vsync = 0;
+	m_hsync = 0;
+
+	m_color_mode = m_s1->read();
+	LOGSETUP("%s: m_color_mode:%02x\n", FUNCNAME, m_color_mode);
+	m_pal = (m_io_monitor-> read() & 1) == 1 ? &m_371x_pal : &m_3111_pal;
+	m_vmode = 0;
+
+	if (m_installed == false)
+	{
+		m_isa->install_device(0x3b0, 0x3bf, read8_delegate( FUNC(isa8_epc_mda_device::io_read), this ), write8_delegate( FUNC(isa8_epc_mda_device::io_write), this ) );
+		m_isa->install_bank(0xb0000, 0xb7fff, "bank_epc", &m_videoram[0]); // Monochrome emulation mode VRAM address
+
+		// This check allows a color monitor adapter to be installed at this address range if color emulation is disabled
+		if (m_color_mode & 1)
+		{
+			m_isa->install_device(0x3d0, 0x3df, read8_delegate( FUNC(isa8_epc_mda_device::io_read), this ), write8_delegate( FUNC(isa8_epc_mda_device::io_write), this ) );
+			m_isa->install_bank(0xb8000, 0xbffff, "bank_epc", &m_videoram[0]); // Color emulation mode VRAM address, but same 32KB areas as there are only this amount on the board
+		}
+		m_installed = true;
+	}
+}
+
+/*
+ * Register Address table from the manual
+ * Ericsson name          MDA mode   CGA mode  Standard name
+ *-------------------------------------------------------------------------------
+ * 6845 Address Registers 0x3b4      0x3d4     wo CRT Index reg
+ * 6845 Data Registers    0x3b5      0x3d5     wo CRT Data reg
+ * Mode Register 1        0x3b8      0x3d8     rw MDA/CGA mode reg (bit 0,1 & 4 incompatible) 
+ * Mode Register 2        0x3bf      0x3df     rw CRT/CPU page reg (incompatible w PCjr only)
+ * Status Register        0x3ba      0x3da     r  CGA/MDA status reg (incompatible)
+ *                                              w EGA/VGA feature ccontrol reg (not used by this board)
+ */
+WRITE8_MEMBER( isa8_epc_mda_device::io_write)
```

Comment:
Either `FOO( bar )` or `FOO(bar)` for macros, not some half-and-half.

Reply:
ok, killed a few old ones too